packages/cloud_firestore/lib/src/blob.dart

```diff
@@ -10,8 +10,18 @@ class Blob {
 
   @override
   bool operator ==(dynamic other) =>
-      other is Blob && DeepCollectionEquality().equals(other.bytes, bytes);
+      other is Blob && other.hashCode == hashCode;
 
   @override
-  int get hashCode => bytes.hashCode;
+  int get hashCode {
+    return hashCodeIteration(0, bytes);
+  }
+
+  int hashCodeIteration(int hashBase, Iterable<int> byteList){
```

Comment:
This might be very slow for large blobs. Is there a better way to calculate the hash based on the blobs actual value?

Reply:
Yes, https://docs.flutter.io/flutter/dart-ui/hashList.html